recipe/meta.yaml

```
@@ -1,44 +1,63 @@
 {% set name = "treelib" %}
-{% set version = "1.6.1" %}
+{% set version = "1.7.1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 1cbfffb2d2b75ccac27d0200cee0507b6fbb0726e0afb9fae017ade5d2ce8788
+  sha256: a2b57ba38290f7da6af97b41de8ff7be98e7a8c00815f95d0782241f77503952
 
 build:
-  noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install --no-deps . -vv
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
   host:
     - python
     - pip
+    - setuptools
+    - wheel
   run:
+    - six
     - python
 
+{% set tests_to_skip = "" %}
+# UnicodeEncodeError: 'charmap' codec can't encode characters in position 7-9: character maps to <undefined>
+{% set tests_to_skip = "not test_show" %}  # [win]
+
 test:
+  source_files:
+    - tests
   imports:
     - treelib
+  commands:
+    - pip check
+    - python -c "from importlib.metadata import version; assert(version('{{ name }}')=='{{ version }}')"
+    - pytest -v -k "{{ tests_to_skip }}" tests
```

```suggestion
    - set PYTHONIOENCODING=utf8  # [win]
    - pytest -v tests
```
This should make the failed test pass.